x/reward/keeper/reward_distribution_test.go

```diff
@@ -322,9 +354,9 @@ func TestKeeper_DistributeRewards(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			// set test reward pool if contains coins
-			if tt.rewardPool.Coins != nil {
+			if tt.rewardPool.CurrentCoins != nil {
```

Comment:
We should add in the assertion that `InitialCoin` is left unchanged in all case
And I realize we should also check that `Coins`/`CurrentCoins`/`RemainingCoins` is decreased by the sum of `tt.WantBalances`

It can be also part of https://github.com/tendermint/spn/issues/502 but I may think it makes sense to add the assertions in this PR

Reply:
I added the `InitialCoin` check.

The content of #502 was giving me some trouble.  I added a check (which is now commented out) but it was causing some of the the tests to fail. Thoughts?

Reply:
```
// TODO: https://github.com/tendermint/spn/issues/502
// assert the module account has no coin left
// accAddr, err := sdk.AccAddressFromBech32(tt.rewardPool.Provider)
// require.NoError(t, err)
// err = bk.SendCoinsFromModuleToAccount(ctx, types.ModuleName, accAddr,
//	sdk.NewCoins(sdk.NewCoin(
//		rewardPool.InitialCoins.GetDenomByIndex(0),
//		sdk.OneInt(),
//	)))
// require.ErrorIs(t, err, sdkerrors.ErrInsufficientFunds)
```
Mmh I don't actually get what we are trying to test here
At the end of the table test body we are burning back the values for each account specified in `WantBalances` to make each test case independent and stateless. We should not from this design send coins to account balances

The idea mentioned above was that we fetch at the beginning of the body `previousInitialCoins` and `previousRemainingCoins` and we assert after `DistributeRewards`
```
previousInitialCoins == newInitialCoins
previousRemainingCoins == newRemainingCoins + sum(wantBalances)
```
I don't think we should transfer balances

Reply:
Got it.  I was trying to find a way to actually check the amount stored in the module account is 0.  As far as I am aware, the only way to check that is to try to send from the module account.  If we send 1 coin from it and get `ErrInsufficientFunds`, this means that the balance has to be 0.